File: npm/src/commands/snippet.ts
import { CLIContext } from '../cli';
import { readConfig, resolveEndpoint } from '../lib/config';
import { error, heading, json as jsonOut, dim } from '../lib/output';

export async function snippet(ctx: CLIContext): Promise<void> {
  const config = readConfig();
  if (!config) {
    error('No .formhandle config found. Run "formhandle init" first.');
    process.exit(1);
  }

  const { domain, endpoint } = resolveEndpoint(config, ctx.flags.domain);

  const actionUrl = `https://api.formhandle.dev/submit/${endpoint.handler_id}`;

  const formHtml = `<form action="${actionUrl}" method="POST">
  <input type="text" name="name" placeholder="Name" required>
  <input type="email" name="email" placeholder="Email" required>
  <textarea name="message" placeholder="Message" required></textarea>
  <button type="submit">Send</button>
</form>`;

  if (ctx.flags.json) {
    jsonOut({
      domain,
      handler_id: endpoint.handler_id,
      action_url: actionUrl,
      form_html: formHtml,
    });
    return;
  }

  heading(`Snippet for ${domain}`);

  console.log(dim('Add this form to your page:'));
  console.log();
  for (const line of formHtml.split('\n')) {
    console.log(`  ${line}`);
  }
  console.log();
}

File: python/src/formhandle/commands/snippet.py
"""Output embed code for your site."""

import sys

from formhandle.config import read_config, resolve_endpoint
from formhandle.output import error, heading, json_out, dim


def run(args) -> None:
    config = read_config()
    if not config:
        error('No .formhandle config found. Run "formhandle init" first.')
        sys.exit(1)

    resolved = resolve_endpoint(config, getattr(args, "domain", None))
    domain = resolved["domain"]
    endpoint = resolved["endpoint"]
    hid = endpoint["handler_id"]

    script_tag = f'<script src="https://api.formhandle.dev/s/{hid}.js"></script>'
    form_html = """<form data-formhandle>
  <input type="text" name="name" placeholder="Name" required>
  <input type="email" name="email" placeholder="Email" required>
  <textarea name="message" placeholder="Message" required></textarea>
  <button type="submit">Send</button>
</form>"""

    if args.json:
        json_out({
            "domain": domain,
            "handler_id": hid,
            "script_tag": script_tag,
            "form_html": form_html,
        })
    else:
        heading(f"Snippet for {domain}")
        print("Add this script tag to your page:\n")
        print(f"  {script_tag}")
        print("\nExample form:\n")
        for line in form_html.split("\n"):
            print(f"  {line}")
        print("\nAttributes:")
        print(f'  data-formhandle-success="…"  {dim("Custom success message")}')
        print(f'  data-formhandle-error="…"    {dim("Custom error message")}')
        print()
